libcloud/compute/drivers/gridscale.py

```diff
@@ -0,0 +1,977 @@
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/ips/{}'
+                                    .format(ip.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def destroy_volume_snapshot(self, snapshot):
+        """
+        Destroy a snapshot.
+
+        :param snapshot: The snapshot to delete.
+        :type snapshot: :class:'.VolumeSnapshot`
+
+        :return: True if the destroy was successful, otherwise False.
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/storages/'
+                                             '{}/snapshots/{}/'
+                                    .format(snapshot.extra['parent_uuid'],
+                                            snapshot.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def ex_destroy_networks(self, network):
+        """
+        Delete network.
+
+        :param network: Network object.
+        :type network: :class:`.GridscaleNetwork`
+
+        :return: ``True`` if destroyed successfully, otherwise ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/networks/{}'
+                                    .format(network.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def delete_image(self, node_image):
+        """
+        Destroy an image.
+
+        :param node_image: Node image object.
+        :type node_image: :class:`.NodeImage`
+
+        :return: True if the destroy was successful, otherwise False
+        :rtype: ``bool``
+
+        """
+        result = self._sync_request(endpoint='objects/templates/{}'
+                                    .format(node_image.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def patch_node(self, node, name=None):
+        """
+        change node name
+
+        :param name: new name
+        :type name: ``str``
+
+        :param node: Server you want to patch
+        :type node: ``str``
+
+        :return: ``True`` or ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(data=json.dumps({'name': name}),
+                                    endpoint='objects/servers/{}'
+                                    .format(node.id),
+                                    method='PATCH')
+        return result.status == 204
+
+    def patch_storage(self, storage, name=None):
+        """
+        modify storage name
+
+        :param storage: storage object
+        :type storage: ``object``
+
+        :param name: rename storage
+        :type name: ``str``
+
+        :return: ``True`` or ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(data=json.dumps({'name': name}),
+                                    endpoint='objects/servers/{}'
```

Comment:
I assume that's a typo and should be `endpoint='objects/storages/{}'`?

Having a test case which also verifies that a particular driver method hits the correct API endpoint also wouldn't hurt :)